src/statemachine.rs

```diff
@@ -319,11 +363,19 @@ impl StateMachine {
             callback.clone(),
             status,
             move |info, selector, status, alive| {
-                let mut dev = match Self::init_and_select(info, &selector, &status, true, alive) {
-                    None => {
-                        return;
-                    }
+                let mut dev = match Self::init_device(info, &selector) {
                     Some(dev) => dev,
+                    None => return,
+                };
+
+                if dev.get_protocol() != FidoProtocol::CTAP2 {
+                    info!("Device does not support CTAP2");
+                    let _ = selector.send(DeviceSelectorEvent::NotAToken(dev.id()));
+                    return;
+                }
```

Comment:
Not really important, but why remove that from `init_device()` and duplicate it here multiple times?
Then all `NotAToken()`-messages would be in the same place.

Reply:
It just seemed logically separate from device initialization. I could go either way.

Reply:
Yes, from a 'functionality separation'-POV, this makes sense. I just slightly dislike the code duplication. But given that both solutions have their drawback, I'm fine with either, too. :)